### ISSUE TITLE
sendMail with tls does not work when I use openssl

### ISSUE THREAD

**Issue by Kozzi11** [vibe-d/vibe.d#2234]
i have this code:

	SMTPClientSettings smtpSettings = new SMTPClientSettings("mail.ipult.cz", 587);
	smtpSettings.authType = SMTPAuthType.plain;
	smtpSettings.username = "noreply@somedomain.com";
	smtpSettings.password = "somepasswd";
	smtpSettings.tlsValidationMode = TLSPeerValidationMode.none;
	smtpSettings.connectionType = SMTPConnectionType.startTLS;
	sendMail(smtpSettings, email);

When I use openssl-1.1 for vibe-d:tls it does not work:

    [main(IM1g) INF] Connecting TLS tunnel:  (0)
[main(IM1g) INF] Connecting TLS tunnel: error:20078044:BIO routines:bio_read_intern:internal error (537362500)

With botan everything works ok

**Comment by Kozzi11** [comment]
When I change `openssl-1.1` to `openssl` and fix linking errors it works ok too

**Comment by jpf91** [comment]
@s-ludwig I think this is likely a problem with new OpenSSL versions: On Fedora 29 with OpenSSL 1.1.1 (released 11.09.2018) I get this `error:20078044:BIO routines:bio_read_intern:internal error (537362500)` with everything TLS related: vibe.d HTTPS server, vibe.d web client for HTTPS websites, SMTPS, ...
@Kozzi11 which OpenSSL version did you use?

If the problem is really the newer OpenSSL library, this will become a serious problem once more distributions update OpenSSL.

Full backtrace:
```
Exception@/home/jpfau/.dub/packages/vibe-d-0.8.4/vibe-d/tls/vibe/stream/openssl.d(381): Connecting TLS tunnel: error:20078044:BIO routines:bio_read_intern:internal error (537362500)
----------------
/home/jpfau/.dub/packages/vibe-d-0.8.4/vibe-d/tls/vibe/stream/openssl.d:381 @safe int vibe.stream.openssl.OpenSSLStream.enforceSSL(int, immutable(char)[]) [0x9e3c1c]
/home/jpfau/.dub/packages/vibe-d-0.8.4/vibe-d/tls/vibe/stream/openssl.d:339 @safe int vibe.stream.openssl.OpenSSLStream.checkSSLRet(int, immutable(char)[]) [0x9e3708]
/home/jpfau/.dub/packages/vibe-d-0.8.4/vibe-d/tls/vibe/stream/openssl.d:167 @safe vibe.stream.openssl.OpenSSLStream vibe.stream.openssl.OpenSSLStream.__ctor(vibe.core.stream.Stream, vibe.stream.openssl.OpenSSLContext, vibe.stream.tls.TLSStreamState, immutable(char)[], vibe.core.net.NetworkAddress, immutable(char)[][]) [0x9e2901]
/home/jpfau/.dub/packages/vibe-d-0.8.4/vibe-d/tls/vibe/stream/openssl.d:672 @safe vibe.stream.openssl.OpenSSLStream vibe.stream.openssl.OpenSSLContext.createStream(vibe.core.stream.Stream, vibe.stream.tls.TLSStreamState, immutable(char)[], vibe.core.net.NetworkAddress) [0x9e49c2]
/home/jpfau/.dub/packages/vibe-d-0.8.4/vibe-d/tls/vibe/stream/tls.d:145 @safe vibe.stream.tls.TLSStream vibe.stream.tls.createTLSStream!(vibe.core.net.TCPConnection).createTLSStream(vibe.core.net.TCPConnection, vibe.stream.tls.TLSContext, vibe.stream.tls.TLSStreamState, immutable(char)[], vibe.core.net.NetworkAddress) [0x8b531b]
/home/jpfau/.dub/packages/vibe-d-0.8.4/vibe-d/http/vibe/http/client.d:630 @safe bool vibe.http.client.HTTPClient.doRequest(scope void delegate(vibe.http.client.HTTPClientRequest), ref bool, bool, std.datetime.systime.SysTime) [0x90c964]
/home/jpfau/.dub/packages/vibe-d-0.8.4/vibe-d/http/vibe/http/client.d:537 @safe bool vibe.http.client.HTTPClient.doRequestWithRetry(scope void delegate(vibe.http.client.HTTPClientRequest), bool, out bool, out std.datetime.systime.SysTime) [0x90c3c9]
/home/jpfau/.dub/packages/vibe-d-0.8.4/vibe-d/http/vibe/http/client.d:509 @safe vibe.http.client.HTTPClientResponse vibe.http.client.HTTPClient.request(scope void delegate(vibe.http.client.HTTPClientRequest)) [0x90c1d8]
/home/jpfau/.dub/packages/vibe-d-0.8.4/vibe-d/http/vibe/http/client.d:103 @safe vibe.http.client.HTTPClientResponse vibe.http.client.requestHTTP(vibe.inet.url.URL, void delegate(scope vibe.http.client.HTTPClientRequest), const(vibe.http.client.HTTPClientSettings)) [0x90ac61]
/home/jpfau/.dub/packages/vibe-d-0.8.4/vibe-d/http/vibe/http/client.d:77 @safe vibe.http.client.HTTPClientResponse vibe.http.client.requestHTTP(immutable(char)[], scope void delegate(scope vibe.http.client.HTTPClientRequest), const(vibe.http.client.HTTPClientSettings)) [0x90a990]

```

**Comment by s-ludwig** [comment]
#2226 supposedly has fixed this, although I don't know all the exact symptoms.

**Comment by s-ludwig** [comment]
In that case, the latest 0.8.5-alpha.2 should work.

**Comment by Kozzi11** [comment]
@s-ludwig @jpf91 #2226 fix it for me (0.8.5-alpha.2 works), so I am closing this

**Comment by jpf91** [comment]
0.8.5-alpha.2 works for me as well. Thanks!
